### PR TITLE
Implicit solver compatibility

### LINKED ISSUE #22: Error with an Implicit solver: no method matching ldiv!
Minimum example
```julia
using DifferentialEquations
using LabelledArrays  # v0.0.4
 
function lv_model!(du, u, p, t)
    du.x = 8 * u.x - 2 * u.x * u.y
    du.y = -4 * u.y + 1 * u.x * u.y
    return du
end

u0 = @LVector [4.0, 6.0] (:x, :y)
tspan = (0.0, 10.0)
prob = ODEProblem(lv_model!, u0, tspan)
sol = solve(prob, Tsit5())  # OK
sol = solve(prob, Rosenbrock23())  # Error
```
Error message:
```
ERROR: MethodError: no method matching ldiv!(::LinearAlgebra.LU{Float64,Array{Float64,2}}, ::LVector{Float64,Array{Float64,1},(:x, :y)})
Closest candidates are:
  ldiv!(::LinearAlgebra.Transpose{#s561,#s560} where #s560<:LinearAlgebra.LowerTriangular where #s561, ::AbstractArray{T,1} where T) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:1217
  ldiv!(::LinearAlgebra.Transpose{#s561,#s560} where #s560<:LinearAlgebra.LowerTriangular where #s561, ::AbstractArray{T,1} where T, ::AbstractArray{T,1} where T) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:1201
  ldiv!(::LinearAlgebra.Transpose{#s561,#s560} where #s560<:LinearAlgebra.UnitLowerTriangular where #s561, ::AbstractArray{T,1} where T) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/LinearAlgebra/src/triangular.jl:1235
  ...
Stacktrace:
 [1] (::LinSolveFactorize{typeof(LinearAlgebra.lu!)})(::LVector{Float64,Array{Float64,1},(:x, :y)}, ::Array{Float64,2}, ::LVector{Float64,Array{Float64,1},(:x, :y)}, ::Bool) at /home/sosiris/.julia/packages/DiffEqBase/nW6r3/src/linear_nonlinear.jl:15
 [2] macro expansion at /home/sosiris/.julia/packages/Parameters/35RKe/src/Parameters.jl:734 [inlined]
 [3] macro expansion at /home/sosiris/.julia/packages/Parameters/35RKe/src/Parameters.jl:733 [inlined]
 [4] perform_step!(::OrdinaryDiffEq.ODEIntegrator{Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType},LVector{Float64,Array{Float64,1},(:x, :y)},Float64,Nothing,Float64,Float64,Float64,Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},ODESolution{Float64,2,Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},Nothing,Nothing,Array{Float64,1},Array{Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},1},ODEProblem{LVector{Float64,Array{Float64,1},(:x, :y)},Tuple{Float64,Float64},true,Nothing,ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing,DiffEqBase.StandardODEProblem},Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType},OrdinaryDiffEq.InterpolationData{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},Array{Float64,1},Array{Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},1},OrdinaryDiffEq.Rosenbrock23Cache{LVector{Float64,Array{Float64,1},(:x, :y)},LVector{Float64,Array{Float64,1},(:x, :y)},Array{Float64,2},Array{Float64,2},OrdinaryDiffEq.Rosenbrock23ConstantCache{Float64,typeof(identity),typeof(identity)},DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},LinSolveFactorize{typeof(LinearAlgebra.lu!)},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2,Tuple{LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)}}},ForwardDiff.DerivativeConfig{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},1},(:x, :y)}}}}},ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},OrdinaryDiffEq.Rosenbrock23Cache{LVector{Float64,Array{Float64,1},(:x, :y)},LVector{Float64,Array{Float64,1},(:x, :y)},Array{Float64,2},Array{Float64,2},OrdinaryDiffEq.Rosenbrock23ConstantCache{Float64,typeof(identity),typeof(identity)},DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},LinSolveFactorize{typeof(LinearAlgebra.lu!)},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2,Tuple{LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)}}},ForwardDiff.DerivativeConfig{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},1},(:x, :y)}}},OrdinaryDiffEq.DEOptions{Float64,Float64,Float64,Float64,typeof(DiffEqBase.ODE_DEFAULT_NORM),typeof(LinearAlgebra.opnorm),CallbackSet{Tuple{},Tuple{}},typeof(DiffEqBase.ODE_DEFAULT_ISOUTOFDOMAIN),typeof(DiffEqBase.ODE_DEFAULT_PROG_MESSAGE),typeof(DiffEqBase.ODE_DEFAULT_UNSTABLE_CHECK),DataStructures.BinaryHeap{Float64,DataStructures.LessThan},DataStructures.BinaryHeap{Float64,DataStructures.LessThan},Nothing,Nothing,Int64,Array{Float64,1},Array{Float64,1},Array{Float64,1}},LVector{Float64,Array{Float64,1},(:x, :y)},Float64}, ::OrdinaryDiffEq.Rosenbrock23Cache{LVector{Float64,Array{Float64,1},(:x, :y)},LVector{Float64,Array{Float64,1},(:x, :y)},Array{Float64,2},Array{Float64,2},OrdinaryDiffEq.Rosenbrock23ConstantCache{Float64,typeof(identity),typeof(identity)},DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},LinSolveFactorize{typeof(LinearAlgebra.lu!)},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2,Tuple{LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)}}},ForwardDiff.DerivativeConfig{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},1},(:x, :y)}}}, ::Bool) at /home/sosiris/.julia/packages/OrdinaryDiffEq/9Kohn/src/perform_step/rosenbrock_perform_step.jl:25
 [5] macro expansion at /home/sosiris/.julia/packages/OrdinaryDiffEq/9Kohn/src/perform_step/rosenbrock_perform_step.jl:25 [inlined]
 [6] solve!(::OrdinaryDiffEq.ODEIntegrator{Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType},LVector{Float64,Array{Float64,1},(:x, :y)},Float64,Nothing,Float64,Float64,Float64,Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},ODESolution{Float64,2,Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},Nothing,Nothing,Array{Float64,1},Array{Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},1},ODEProblem{LVector{Float64,Array{Float64,1},(:x,:y)},Tuple{Float64,Float64},true,Nothing,ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing,DiffEqBase.StandardODEProblem},Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType},OrdinaryDiffEq.InterpolationData{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},Array{Float64,1},Array{Array{LVector{Float64,Array{Float64,1},(:x, :y)},1},1},OrdinaryDiffEq.Rosenbrock23Cache{LVector{Float64,Array{Float64,1},(:x, :y)},LVector{Float64,Array{Float64,1},(:x, :y)},Array{Float64,2},Array{Float64,2},OrdinaryDiffEq.Rosenbrock23ConstantCache{Float64,typeof(identity),typeof(identity)},DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},LinSolveFactorize{typeof(LinearAlgebra.lu!)},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2,Tuple{LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)}}},ForwardDiff.DerivativeConfig{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},1},(:x, :y)}}}}},ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},OrdinaryDiffEq.Rosenbrock23Cache{LVector{Float64,Array{Float64,1},(:x, :y)},LVector{Float64,Array{Float64,1},(:x, :y)},Array{Float64,2},Array{Float64,2},OrdinaryDiffEq.Rosenbrock23ConstantCache{Float64,typeof(identity),typeof(identity)},DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},LinSolveFactorize{typeof(LinearAlgebra.lu!)},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2,Tuple{LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Nothing},Float64},Float64,2},1},(:x, :y)}}},ForwardDiff.DerivativeConfig{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},LVector{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.TimeGradientWrapper{ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},LVector{Float64,Array{Float64,1},(:x, :y)},Nothing},Float64},Float64,1},1},(:x, :y)}}},OrdinaryDiffEq.DEOptions{Float64,Float64,Float64,Float64,typeof(DiffEqBase.ODE_DEFAULT_NORM),typeof(LinearAlgebra.opnorm),CallbackSet{Tuple{},Tuple{}},typeof(DiffEqBase.ODE_DEFAULT_ISOUTOFDOMAIN),typeof(DiffEqBase.ODE_DEFAULT_PROG_MESSAGE),typeof(DiffEqBase.ODE_DEFAULT_UNSTABLE_CHECK),DataStructures.BinaryHeap{Float64,DataStructures.LessThan},DataStructures.BinaryHeap{Float64,DataStructures.LessThan},Nothing,Nothing,Int64,Array{Float64,1},Array{Float64,1},Array{Float64,1}},LVector{Float64,Array{Float64,1},(:x, :y)},Float64}) at /home/sosiris/.julia/packages/OrdinaryDiffEq/9Kohn/src/solve.jl:349
 [7] #__solve#202(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::ODEProblem{LVector{Float64,Array{Float64,1},(:x, :y)},Tuple{Float64,Float64},true,Nothing,ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing,DiffEqBase.StandardODEProblem}, ::Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType}, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /home/sosiris/.julia/packages/OrdinaryDiffEq/9Kohn/src/solve.jl:7
 [8] __solve at /home/sosiris/.julia/packages/OrdinaryDiffEq/9Kohn/src/solve.jl:6 [inlined] (repeats 3 times)
 [9] #solve#442(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::ODEProblem{LVector{Float64,Array{Float64,1},(:x, :y)},Tuple{Float64,Float64},true,Nothing,ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing,DiffEqBase.StandardODEProblem}, ::Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType}) at /home/sosiris/.julia/packages/DiffEqBase/nW6r3/src/solve.jl:39
 [10] solve(::ODEProblem{LVector{Float64,Array{Float64,1},(:x, :y)},Tuple{Float64,Float64},true,Nothing,ODEFunction{true,typeof(lv_model!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing,DiffEqBase.StandardODEProblem}, ::Rosenbrock23{0,true,LinSolveFactorize{typeof(LinearAlgebra.lu!)},DataType}) at /home/sosiris/.julia/packages/DiffEqBase/nW6r3/src/solve.jl:27
 [11] top-level scope at none:0
```

### PR DESCRIPTION
**Pull request by YingboMa**
Fixes #22

### CODE REVIEW

File: test/diffeq.jl [hunk 1]
```diff
@@ -14,6 +14,7 @@ u0 = LorenzVector(1.0,0.0,0.0)
 p = LorenzParameterVector(10.0,28.0,8/3)
 tspan = (0.0,10.0)
 prob = ODEProblem(f,u0,tspan,p)
+@test_broken sol = solve(prob,Rosenbrock23())
```
Comment (ChrisRackauckas) [inline test/diffeq.jl]: why is this one broken? It should just use the StaticArray LU/QR dispatches?

Comment (YingboMa) [reply test/diffeq.jl]: It is broken because ForwardDiff tries to do `vector_mode_jacobian` and it uses `MArray`, but `MArray` doesn't have labels.

Comment (ChrisRackauckas) [reply test/diffeq.jl]: ForwardDiff doesn't internally use it as a SArray? And I'm surprised this doesn't use `similar` or `similar_type` to generate the internals, and thus keep the labels?

Comment (YingboMa) [reply test/diffeq.jl]:
```julia
julia> u0 |> typeof
SLArray{Tuple{3},1,(:x, :y, :z),Float64}

julia> similar(u0)
3-element MArray{Tuple{3},Float64,1,3}:
 3.3e-322
 6.9339732699493e-310
 0.0
```

Comment (ChrisRackauckas) [reply test/diffeq.jl]: We need an `MArray` version to `similar` to then... there must be an easy way.

**Pull request merged**